chain/client/src/view_client.rs

```diff
@@ -505,6 +528,62 @@ impl ViewClientActor {
         cache.push_back(now);
         true
     }
+
+    fn get_chain_info(&mut self) -> ChainInfo {
```

Comment:
In general I don't quite like how get_chain_info needs to get through view_client. Most of the information there don't change at all (genesis_id, tracked_shards, and archival) and don't need to be requested every time. We can add these in `PeerManagerState` and initialize them when we initialize `PeerManagerActor`. The only thing that changes is `height` and 'this_epoch_info` and `next_epoch_info`, and we can make ClientActor to send an update to PeerManager to update them at every block.

Reply:
It is called whenever a new peer connects (very rarely) and whenever SyncAccountsData is received (every few seconds). I'm adding rate limiting for the sender side in the next PR, throttling on the receiver side will be done next.

I agree that the updates from view client should be rather pushed than pulled in this case (it would simplify a few things on the PeerManager side). Do you want me to change that in this PR? I don't know exactly all the places in client where the epoch & height can actually change, so can you help me with that? In fact if you think it would be better, I can extract the epoch accounts from the ChainInfo so that client would need to send an update only when the epoch changes (i.e. the height fetching would stay in pull mode, as it was so far).